api/src/gmsa_service.cpp

```diff
@@ -390,6 +392,16 @@ class CredentialsFetcherImpl final
                             // get credentialspec contents:
                             Aws::Auth::AWSCredentials creds =
                                 get_credentials( accessId, secretKey, sessionToken );
+
+                            bool isObjectValid = check_file_size_s3(results[0], region, creds,
+                                                                     false );
+                            if(!isObjectValid)
+                            {
+                                err_msg = "ERROR: invalid object for credentialspec in s3";
```

Comment:
Does this message specify if the file size is more

Reply:
invalid file should be fine